protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```diff
@@ -120,10 +120,14 @@ pub struct StagedPhash {
 }
 
 impl StagedPhash {
-    pub fn into_set_p_hash(&self, channel_id: u32) -> SetNewPrevHash<'static> {
+    pub fn into_set_p_hash(
+        &self,
+        channel_id: u32,
+        new_job_id: Option<u32>,
```

Comment:
Why we need to have an option here? What changed? Why after these PR the Option wasn't needed and now it is?

Reply:
So the problem is here:
bosminer::client::stratum_v2: incoming=NewMiningJob { channel_id: 1, job_id: 2, future_job: true) }

bosminer::client::stratum_v2: incoming=SetNewPrevHash { channel_id: 1, job_id: 0, }

before this change when I just swapped the messages the SNPH's job_id didn't match the NMJ message that proceeded it. This change injects the NMJ's job_id into the StagedPhash so that when the SNPH message goes out the job_ids match.